protocol/encoding.go

```diff
@@ -0,0 +1,83 @@
+// Defines methods/functions to encode/decode messages between client and server.
+// Currently this module supports JSON marshal/unmarshal only.
+// Protobuf would be supported in the feature.
+
+package protocol
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+var b64en = base64.StdEncoding.EncodeToString
+
+func MarshalTemporaryBinding(tb *merkletree.TemporaryBinding) ([]byte, error) {
+	return json.Marshal(tb)
+}
+
+func MarshalSTR(str *merkletree.SignedTreeRoot) ([]byte, error) {
+	policies, err := json.Marshal(str.Policies)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&struct {
+		TreeHash        []byte
+		Epoch           uint64
+		PreviousEpoch   uint64
+		PreviousSTRHash []byte
+		Policies        json.RawMessage
+		Signature       []byte
+	}{
+		TreeHash:        str.Root(),
+		Epoch:           str.Epoch,
+		PreviousEpoch:   str.PreviousEpoch,
+		PreviousSTRHash: str.PreviousSTRHash,
+		Policies:        policies,
+		Signature:       str.Signature,
+	})
```

Comment:
Whoops, I see they are already in a separate encoding.go file and you won't need them anywhere else. So it's fine! Sorry for the noise, too.